### DISCUSSION TITLE
Enhancing CF Community Collaboration on Zenodo: Review and Publication Workflow

### DISCUSSION THREAD

**Discussion by cofinoa** [cf-convention/discuss#326] [Comments and ideas for changing CF]
Dear @cf-convention/info-mgmt,

With the recent establishment of the CF Community on Zenodo, I believe we should discuss how we can improve the management of **releases and community-driven publication processes** on this platform. 

Since Zenodo file depositions are curated and cannot be modified after publication, it is important to have a **manual review step** in place before finalizing any deposit. Unlike GitHub releases, which are more flexible, Zenodo requires extra caution to avoid any versioning issues or accidental errors.

I propose modifying the current workflow by integrating a **draft stage** in the Zenodo deposit process, triggered by a GitHub Action when a new release is published on GitHub. This draft can then be reviewed and, once approved, published to Zenodo. During this review process, we can ensure that all relevant materials—such as rendered HTML and PDF documents—are included, alongside the source files. An added benefit of this draft stage is the generation of a **temporary DOI**, which can be used in documents before final publication.

Moreover, to encourage broader **community participation**, we can leverage the **Zenodo Community** feature, which allows multiple contributors to take on roles such as Reader, Curator, Manager, or Owner. This would decentralize the publication process and make it more inclusive, allowing more community members to participate in the review and approval of deposits. 

I have set up a **CF Conventions Zenodo Community** for testing:  
[CF Conventions Zenodo Community](https://zenodo.org/communities/cfconventions)

For more information on how Zenodo communities work, please see the documentation here:  
[Zenodo Community Documentation](https://help.zenodo.org/docs/communities/)

I have already invited @ethanrd as an owner of this community to help with testing.

We can also set up a **review process** for submissions to the community, so deposits can be checked by multiple community members before publication:  
[Zenodo Submission Review](https://help.zenodo.org/docs/communities/review-submissions/)

I would like your input on this idea and whether it is worth drafting a GitHub Action to initiate this workflow and test it in the community.

Note: Zenodo provides a sandbox environment for testing deposit workflows:  
> Want to test Zenodo? Use our sandbox environment at [Zenodo Sandbox](https://sandbox.zenodo.org/)

**Comment by cofinoa** [comment]
I have added an "about" page:  
https://zenodo.org/communities/cfconventions/about  
and a "curation policy" page:  
https://zenodo.org/communities/cfconventions/curation-policy  

Please take a moment to review them and provide any feedback or suggestions for improvement.

**Comment by cofinoa** [comment]
Dear @cf-convention/info-mgmt,

As suggested by @ethanrd:

> Perhaps the info-mgmt team (or some subset) should be added to the Zenodo community. Probably with the Manager role? Though one or two more Owners might be good.

I have invited the following members of the **CF Community on Zenodo** to be Managers:
- [X] @japamment
- [X] @DocOtak
- [X] @erget
- [ ] @castelao
- [X] @sadielbartholomew
- [X] @davidhassell
- [X] @efisher008
- [X] @JonathanGregory

Please accept the invitation and confirm that I have invited the correct Zenodo users.

I could not find the Zenodo usernames for @feggleton and @RosalynHatcher. If you haven’t signed up yet, you can use your **GitHub identity** to sign up and log in to Zenodo.

**Reply by sadielbartholomew** [reply]
Accepted, thanks for setting this up Antonio.

**Comment by cofinoa** [comment]
# Integrate GitHub Release Workflow with Zenodo for DOI Generation in cf-conventions Document Releases

**Description:**

This proposes an enhancement to the cf-conventions document release workflow by integrating GitHub and Zenodo. The goal is to automate DOI creation and updating so that each new release includes an up-to-date DOI.

## Summary of Proposed Workflow

To ensure each release includes a valid DOI, the process will involve drafting, updating, and publishing a Zenodo record in synchronization with GitHub release actions. This draft workflow is currently under development in this branch:
https://github.com/cofinoa/cf-conventions/tree/zenodo-upload

### Steps for Enhanced Release Workflow

1. **Draft Zenodo Record and Mint Temporary DOI**:
   - When a new release is initiated on GitHub, automatically create a draft of the release on Zenodo.
   - Mint a temporary DOI for the draft version, allowing the DOI to be referenced in the conventions document.

2. **Update GitHub Repository with Zenodo DOI**:
   - Insert the newly minted DOI into the conventions document in the GitHub repository, ensuring it’s correctly embedded in the release version.

3. **Finalize Release and Synchronize with Zenodo**:
   - Complete and publish the release on GitHub, triggering the final upload to Zenodo.
   - Review and publish the draft on Zenodo, finalizing the DOI assignment.

**Comment by JonathanGregory** [comment]
Dear Antonio @cofinoa 

Many thanks for this proposal. Since you have a definite scheme in mind, I think you should open a new `GitHub` issue in the conventions repo, and we'll discuss and agree it there. You could use the same text. Unfortunately you can't do that with the "Create issue from discussion" button, because it will be in a different repo.

Best wishes

Jonathan

**Reply by cofinoa** [reply]
Thanks, @JonathanGregory. Navigating cross-references between issues and discussions across multiple repositories can sometimes be a bit overwhelming for me!

**Reply by JonathanGregory** [reply]
Yes, I sympathise. After CF 1.12 is concluded, I'd like to initiate a discussion about managing discussions. It gets confusing!

**Comment by JonathanGregory** [comment]
In fact you could propose it in the existing [issue 513](https://github.com/cf-convention/cf-conventions/issues/513).

**Comment by cofinoa** [comment]
Dear @cf-convention/info-mgmt,  

I have started the publication process for the **v1.12** release on Zenodo SANDBOX. Before this, I manually published the old versions, from **v1.0** through **v1.11**, to ensure completeness.  

Publishing the new version **v1.12** is still a manual process, particularly the steps for updating `version.adoc`, `zenodo.json`, and `CITATION.cff`. These updates are made once the new version on Zenodo is drafted, and a DOI pre-reservation has been created.  

All versions are now available in the Zenodo SANDBOX collection:  
https://sandbox.zenodo.org/records/138825  

Additionally, I have added this collection to the **CF Community SANDBOX**:  
https://sandbox.zenodo.org/communities/cfconventions-sandbox  

### Issues/Comments Found While Curating Old Versions:
- The names for the conformance documents are inconsistent and include the version number in the file name:  
  https://sandbox.zenodo.org/records/138826  
- The file format for the conformance documents is primarily HTML, except for the latest version, **v1.11**, which is in PDF.  
- The conformance file naming convention changed in **v1.8**, but the format remained HTML:  
  https://sandbox.zenodo.org/records/138834  
- Version **v1.7** is tricky due to multiple releases (**1.7.0**, **1.7.2**, and **1.7.3**), with **1.7.3** being the last deposit for **v1.7**.  
- I have added some additional related identifiers, pointing to HTML and PDF artifacts on the CF webpage.  
- From **v1.6** onward, I included the `is derived from` relation, pointing to the corresponding tag in the GitHub repository:  
  https://sandbox.zenodo.org/records/138832  

### Next Steps:
**Note:** I have drafted the **v1.12** release:  
https://sandbox.zenodo.org/uploads/138838  

You will need to sign up for the Zenodo SANDBOX. If you do, please let me know, and I will invite your user ID to the **CF Community** in the Zenodo SANDBOX.  

- [ ] @japamment  
- [ ] @DocOtak  
- [ ] @erget  
- [ ] @castelao  
- [X] @sadielbartholomew  
- [ ] @davidhassell  
- [ ] @efisher008  
- [ ] @JonathanGregory 
- [X] @ethanrd  

**I need your feedback and suggestions to proceed with the actual curation in the "real" ZENODO.**

**Comment by sadielbartholomew** [comment]
Hi @cofinoa, please can you let us know more specifically how to sign up, regarding:

> You will need to sign up for the Zenodo SANDBOX. If you do, please let me know, and I will invite your user ID to the CF Community in the Zenodo SANDBOX.

I have signed into Zenodo but I can't see anywhere to sign up to the sandbox. I suspect you might have to invite people to join it, unless I am misunderstanding.

**Reply by cofinoa** [reply]
@sadielbartholomew, I have invited you as manager to the CF Metadata Conventions in the ZENODO SANDBOX

But records in the CF Community - SANDBOX are public:
https://sandbox.zenodo.org/communities/cfconventions-sandbox/

Also the CF Conventions documents record it's public: 
https://sandbox.zenodo.org/records/138825

**Reply by sadielbartholomew** [reply]
OK, thanks Antonio. I've accepted the invitation.

**Comment by JonathanGregory** [comment]
Dear Antonio @cofinoa 

I have looked at the conventions and conformance documents archived in [this list in the CF Zenodo sandbox](https://sandbox.zenodo.org/search?q=parent.id%3A138825&f=allversions%3Atrue&l=list&p=1&s=10&sort=version) for versions 1.0 to 1.11 inclusive. As far as I can see, they are the same as the conventions PDF and the conformance document (PDF or HTML) on the [Conventions page](https://cfconventions.org/conventions.html). Thanks for assembling this set of documents carefully! I think we're good to go! :rocket:

I believe I have found one mistake in the dates: version 1.6 says it was published on 5 Dec 2011, but Zenodo has 5 Dec 2012 in the metadata. Also, there's a curiosity about the date of version 1.0, which claims to have been published on 28 Oct 2003. However, the version we have got is not that version, because apparently it was updated without changing the version numbers, as you can see in its history. The last change made in that document is dated 21 Mar 2006, which could arguably be given as the date of publication instead of 28 Oct 2003. This is really only of historical interest of course. The next update was on 17 Jan 2008, when the version number was changed to 1.1. It is interesting to note that there was no announcement on the CF email list of the new release! From version 1.1 onwards, the date of publication is the same as or later than the last revision that appears in its history.

Best wishes

Jonathan

**Reply by cofinoa** [reply]
> I believe I have found one mistake in the dates: version 1.6 says it was published on 5 Dec 2011, but Zenodo has 5 Dec 2012 in the metadata

Yes, you’re right—it was a mistake.  

> The last change made in that document is dated 21 Mar 2006, which could arguably be given as the date of publication instead of 28 Oct 2003. This is really only of historical interest of course.

I see your point. I’ve updated the publication date to `2006-03-21`, based on the last change in the document.  

This brings up a similar situation I encountered with **version 1.7**. For this version, I chose the publication date as `2017-08-15`, which corresponds to tag `1.7.3` in the GitHub repository:  
https://github.com/cf-convention/cf-conventions/tree/1.7.3  
This also aligns with the version labelled as 1.7 on the CF Conventions webpage.  

---

### Additional Notes:
- **Versioning Scheme:**  
  I think we should open a broader issue or discussion (if not already done) to clarify the versioning scheme across the repository release/tags, document versioning, and the version used in the `Conventions` attribute. Would the ongoing discussion in [cf-convention/discussions/321](https://github.com/cf-convention/discuss/discussions/321) be the best place for this?  

- **Specific Issues with Version 1.7:**  
  - **Authors:**  
    - "Jon Blower" appears as the first "additional author" in the metadata, whereas other additional authors are appended to the end of the list.  
    - "Randy Horne" is not listed as an author in the document header but appears in the "About the authors" section. I’ve included "Randy Horne" as author in the Zenodo metadata.

- **Issue with Version 1.3:**  
  The CF webpage link for the **version 1.3** HTML document points to version 1.3, but its contents correspond to version 1.2:  
  https://cfconventions.org/Data/cf-conventions/cf-conventions-1.3/build/cf-conventions.html  

Let me know your thoughts on these points.

**Comment by JonathanGregory** [comment]
Dear Antonio

Thanks for these additional points.

* I think you made the correct choice for 1.7. That was when the document was converted to AsciiDoc, and it took a few attempts. Is 1.7.3 the latest one?

* [Discussion 321](https://github.com/cf-convention/discuss/discussions/321) is specifically about the "working" version. I'm not clear what you would like to discuss. If it's not the same matter, I suggest starting a new Discussion about it.

* Jon Blower was added as an additional author in version 1.7. I expect he was put at the front of the additional author list by someone assuming they are alphabetical. He should instead be after Martin Raspaud. I have opened conventions defect [issue 573](https://github.com/cf-convention/cf-conventions/issues/573) to correct this and I will merge it tomorrow if no-one objects.

* Randy Horne should have been added as an additional author in version 1.7 as well, as decided in [trac ticket 102](https://cfconventions.org/Data/Trac-tickets/102.html). They're added sequentially, as you remark, so he should have been included in 1.7 after Martin Raspaud, as he is in 1.8 onwards.

* It seems that the single-HTML file in the 1.3 directory is the wrong file, I agree. The multi-HTML one is OK though, so the simplest solution is to remove the link to the erroneous document on the Conventions page. I have created a branch to fix this but I'm having problems with making a PR just now, so I'll do it in the morning.

Best wishes

Jonathan

**Reply by JonathanGregory** [reply]
Regarding the last point, about links on the Conventions page, I have just opened [website issue 565](https://github.com/cf-convention/cf-convention.github.io/issues/565) to fix it and make a couple of other improvements in line with the Zenodo information.

**Reply by cofinoa** [reply]
- Yes, for version 1.7 the last one is 1.7.3.
- I will follow-up and see if a new issue needs to be open
- I see. Is it OK to fix the authors order, in the Zenodo metadata, not the document itself, from version 1.7 onwards?.
- As previous bullet, it was "fixed" in Zenodo metadata for version 1.7, but not the document itself.
- OK

**Reply by JonathanGregory** [reply]
Yes, I think it would be appropriate to correct the order of additional authors in the Zenodo metadata, even though that means it disagrees with the document, for versions 1.7&ndash;1.11. Thanks.

**Comment by cofinoa** [comment]
## 🎉 Curation Announcement: CF Convention Versions on Zenodo 🚀

Dear @cf-convention/info-mgmt team,  

I’m delighted to announce that **all versions of the CF Convention (1.0 to 1.11)** have been successfully uploaded and curated in Zenodo! 📚✨  

This DOI represents all versions and will always resolve to the latest one:  
[![DOI](https://img.shields.io/badge/DOI-10.5281%2Fzenodo.14274886-blue)](https://doi.org/10.5281/zenodo.14274886)  

Thank you all for your incredible efforts in making this milestone possible! 🙌👏  

🚀 Here's to preserving the legacy of the CF Convention and making it accessible for future generations of CF'ers! 🌟  

Best regards,   

Antonio

**Reply by sadielbartholomew** [reply]
Brilliant stuff, thanks to everyone involved. I got a notification from ORCID that it had automatically been added to my profile, which implies the whole workflow is working seamlessly (or at least, to those not doing some hard work in the background!). Happy DOI day everyone :partying_face: 

![Screenshot_20241204_130109](https://github.com/user-attachments/assets/98f207a9-e27b-4f1e-8710-e69309bfb7a7)

**Comment by JonathanGregory** [comment]
Many thanks to you, Antonio, for your hard work on this.

**Comment by ethanrd** [comment]
Yes, thank you Antonio! Great work.

Does this means that the earlier concept and v1.11 DOIs created with the Zenodo/GitHub integration should not be used any more?
- Concept DOI: https://doi.org/10.5281/zenodo.11288137
- DOI for Version 1.11: https://doi.org/10.5281/zenodo.11288138

Is there a way to deprecate old DOIs? Perhaps we should add an Alternate Identifier referencing the new DOI. Or a Related Work with the `isSupplementedBy` relationship referencing the new DOI.

**Reply by DocOtak** [reply]
If there isn't a way to do what you want via the web interface, it might be possible if you reach out to the zenodo support folks.

**Reply by larsbarring** [reply]
Typically it is not possible (and zenodo will not allow) a doi to be deleted/removed/etc. Exceptions are legal requirements, copyright infringements and other formal reasons. What we however can do is to create a "thombstone page", as they are called, to explain why the doi resource is no longer available and instead point to the new version 1.11.

Speaking of version numbers, it is great that the version number on zenodo is in phase with the actual version of the conventions document. But this is not something we can expect to always be the case. I believe there was a case sometime back when there actually was a minor update 1.X.1 for some technical reason. A similar situation will forever bring the zenodo version number out of phase with the CF version. 

What can we do? 
-- For the existing published versions we change the metadata field `Title` to also state the version number (this can be done after publication)
-- Starting with version 1.13 do the same, and also include the version number in the uploaded file name. Then we might want to consider to do the same in the versions published at the website.

Another thought, do we want to have a `zenodo` label to tag discussions and issues specifically related to zenodo?

**Reply by cofinoa** [reply]
@ethanrd, can you submit the "deprecated" record to the Zenodo CF Community? 😉 Then we can assume it as the CF Community's responsibility.  

I agree with @larsbarring's suggestion regarding **tombstone pages** as a practical solution for handling deprecated DOIs. This approach ensures that users are directed to the correct resources without violating Zenodo's policies. FYI, see: [https://doi.org/10.5281/zenodo.1098445](https://doi.org/10.5281/zenodo.1098445)  

Also, we can restrict the contents of the record and annotate the abstract with flashing ⚠️ emojis to alert users about the deprecation.  

### 🔢 Regarding **version numbers**:  
- I completely agree that aligning Zenodo version numbers with the CF document versions is ideal but not always feasible, especially in scenarios like minor updates (e.g., CF 1.7, where the GitHub release version is v1.7.3) or the special case for 1.0, where revisions were made on different dates but the version was kept as v1.0.  
- **Standardize Versioning**: Open a new discussion or continue in #321 to clarify repository versioning schemes. I need to understand how CF standard versioning works, particularly when it is used in the `Conventions` attribute.  

### 🚀 Starting with **v1.13**:  
- Including the version number in the uploaded filenames makes sense, as it aligns the Zenodo records more closely with the CF website.  
- I agree that we should discuss whether to adopt this filename convention across the CF website for consistency among CF document artifacts.  

Finally, I think adding a **Zenodo label** for GitHub issues and discussions is a great idea. 🏷️ This will help us track Zenodo-related tasks and decisions more effectively.

**Reply by ethanrd** [reply]
@cofinoa - I just submitted the old Zenodo record for v1.11 to the CF community. The concept DOI still resolves to the old v1.11 record so I don't think we have to do anything for that one. I assume since the Zenodo/GitHub link was broken it will continue to point to the v1.11 record.

The [example tombstone page you gave](https://doi.org/10.5281/zenodo.1098445) is, I believe, from an official Zenodo removal of a resource/record. I'm not sure we will be able to remove the resource/record. Instead we may just have to change the metadata to indicate that the DOI should no longer be used. The [Zenodo policy page](https://about.zenodo.org/policies/) mentions tombstone pages but only in terms of official removal for policy violation and such.

**Reply by larsbarring** [reply]
@ethanrd  wrt thombstone pages, am not sure that it as limited as limited as you say. The [zenodo policy](https://about.zenodo.org/policies/) states
> Withdrawal: If the uploaded research object must later be withdrawn, the reason for the withdrawal will be indicated on a tombstone page, which will henceforth be served in its place. Withdrawal is considered an exceptional action, which normally should be requested and fully justified by the original uploader. In any other circumstance reasonable attempts will be made to contact the original uploader to obtain consent. The DOI and the URL of the original object are retained.

What that mean in practice I do not know, and maybe it is more bureaucratic work than warranted, and the more pragmatic way forward is as you say.

**Reply by cofinoa** [reply]
@ethanrd I have accepted the request to include it as part of the **Zenodo CF Conventions Community**.  

I have edited the record:
https://zenodo.org/records/11288138

and made the following changes:  

### **Metadata Changes**  

**1. Title**  
- **Updated Title**:  
  ```
  [DEPRECATED] NetCDF Climate and Forecast (CF) Metadata Conventions
  ```
- Added the **[DEPRECATED]** annotation at the beginning of the original title to make it clear in search results and within the Zenodo UI that this record is no longer valid.  

**2. File Access**  
- **Restricted access** to the files as they contained **source files from the GitHub repository** rather than the intended curated artifacts.  
- The metadata remains public, but the files themselves are no longer accessible.  

---

**3. Related Identifiers**  
- Added a formal relationship to link this deprecated record to the correct record.  
- **Relation Type**: **isObsoletedBy**  
- **DOI**: [10.5281/zenodo.11288138](https://doi.org/10.5281/zenodo.11288138)  

**4. Abstract/Description**  
- Added a **bold, red alert** at the top to clearly indicate that the record has **restricted access** and explain why.  
- **Message Added**:  
  ```html
  <h1>⚠️ Important Notice: This record was published in error.</h1>
  <h3>This record was mistakenly published as part of the GitHub integration release process. The uploaded files were source files 
  from the repository, not the intended curated artifacts for Zenodo.</h3>
  <h3>As a result, access to the files has been restricted to avoid confusion and ensure proper curation. The correct version of the curated document can be found at&nbsp;<a href="https://doi.org/10.5281/zenodo.11288138">DOI: 10.5281/zenodo.11288138</a></h3>
  ```


These changes ensure that users are clearly informed about the status of this record, understand why the files are restricted, and are provided with a direct link to the correct version.  

**Note:** This approach serves as an initial step to provide clarity and transparency about the status of this record. We can later evaluate if a **Tombstone page** should be requested from Zenodo support to more formally mark the record as deprecated.  

Let me know if there are any additional changes or improvements you'd like to see.

**Comment by cofinoa** [comment]
Dear @cf-convention/info-mgmt,  
I am glad to share the drafted upload of the **CF Conventions v1.12** on Zenodo. 🎉  

### 🔎 Review the Draft:  
You can access and review the drafted upload using the link below:  
[Zenodo Draft for CF Conventions v1.12](https://zenodo.org/records/14275599?preview=1&token=eyJhbGciOiJIUzUxMiJ9.eyJpZCI6IjkzYmU2YzZmLTA4YWItNDE2Mi1iZDU3LWUyZjc4NjM2YWM0OSIsImRhdGEiOnt9LCJyYW5kb20iOiJmN2E2OTY5MTZhMjJiYjEyNGY4ZjZkMWE0MGQxZWE1MSJ9.zuNEfpfqcUWEbL3j7xRPFW1BndbYIs6XExDNvD36Sd-5LN7e3VbCtf2qsypXzYeTwTLDsLeNegPJMzhVGJIQhg)  

Additionally, I have updated the assets on the release webpage:  
[CF Conventions v1.12 Release Webpage](https://github.com/cf-convention/cf-conventions/releases/tag/v1.12.0)  

### ✅ Key Updates:
- **Canonical Source:** I assume that the CF Website is the "canonical" source, meaning the authoritative reference for filenames and content. I have respected the filenames from it in this release.  
- **Zenodo Curation:** I uploaded the PDF artifacts only to the Zenodo record, using the same filenames as on the canonical source (i.e., the CF Website).  

### 📝 Notes:
- Historically, the CF Website has been considered the canonical source for Zenodo records from **v1.0** to **v1.10**.  
- This practice was not followed with **v1.11**. 🤦‍♂️ (We’ll need to address this inconsistency in future discussions.)  

### 📢 Feedback Needed:  
Please take a moment to review the draft and provide any feedback or raise any issues. Your input is essential for finalizing the upload.  

Thank you for your support! 😊  

### 📝 Notes:
- The CF Website has been considered the canonical source for Zenodo records from **v1.0** to **v1.10**.  
- This practice was not followed with **v1.11**, raising questions about consistency and standardization moving forward.  

### 📢 Feedback Needed:  
Please take a moment to review the draft and provide any feedback or raise any issues. Your input is essential for finalizing the upload.  

Thank you for your support! 😊

**Reply by cofinoa** [reply]
**Dear @cf-convention/info-mgmt,**

After waiting a prudent amount of time to receive comments on the drafted upload of the CF Conventions 1.12 to Zenodo, and not receiving any revisions or modifications to the draft, I am pleased to announce that I am now releasing it definitively. 🎉

As a reminder, the files on Zenodo are immutable, but the metadata can be updated if required.

Thank you all for your feedback and support throughout this process.

Ping to @cf-convention/committees @cf-convention/gov-panel

**Reply by cofinoa** [reply]
Done!
Eaton, B., Gregory, J., Drach, B., Taylor, K., Hankin, S. et al. (2024). NetCDF Climate and Forecast (CF) Metadata Conventions (1.12). CF Community. https://doi.org/10.5281/zenodo.14275599

**Comment by davidhassell** [comment]
Hi Antonio,

The documents on the GitHub release look fine to me. As do both of the documents, and their metadata, on the draft Zenodo release.

Thanks!

**Comment by JonathanGregory** [comment]
This looks fine to me, Antonio. Thanks very much.

**Comment by JonathanGregory** [comment]
This discussion has come to a successful conclusion, thanks to Antonio, so I'm closing it.
